README.md

```diff
@@ -28,10 +26,132 @@ To generate organization-wide token:
 3. Select the application and choose generate token.
 4. Select `Administer organizations`, `Adminster repositories`, `Create Repositories` permissions.
 
-### Create a Component
+## General purpose image repository
+
+### Requesting image repository
+
+To request an image repository one should create `ImageRepository` custom resource:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+  name: imagerepository-sample
+  namespace: test-ns
+```
+As a result, a public image repository `quay.io/my-org/test-ns/imagerepository-sample` will be created.
+When `status.state` is set to `ready`, the image repository is ready for use.
+Additional information about the image repository one may obtain from the `ImageRepository` object `status`:
+```yaml
+apiVersion: appstudio.redhat.com/v1alpha1
+kind: ImageRepository
+metadata:
+  name: imagerepository-sample
+  namespace: test-ns
+spec:
+  image:
+    name: test-ns/imagerepository-sample
+    visibility: public
+status:
+  credentials:
+    generationTimestamp: "2023-08-23T14:56:41Z"
+    push-robot-account: test_ns_imagerepository_sample_101e4e2b63
+    push-secret: test-ns-imagerepository-sample-101e4e2b63
+  image:
+    url: quay.io/my-org/test-ns/imagerepository-sample
+    visibility: public
+  state: ready
+```
+where:
+  - `push-robot-account` is the name of quay robot account in the configured quay organization with write premissions to the repository.
+  - `push-secret` is a `Secret` of dockerconfigjson type that contains image repository access token with write permissions.
```

Comment:
Would be good to make "access token" more specific. Is it the push robot account token?